### PR TITLE
Remove lists:reverse from yz_solrq_helper

### PR DESCRIPTION
**Pull request by JeetKunDoug**
by using list comprehension instead of lists:foldl and "un-reversing" order of ops in certain cases.

Depends on basho/riak_core#837 for dialyzer to pass.

### CODE REVIEW

File: src/yz_solrq_helper.erl [hunk 1]
```diff
@@ -222,41 +222,51 @@ update_solr(Index, LI, Entries) ->
...
+    BProps = riak_core_bucket:get_bucket(Bucket),
+    Obj = yz_kv:maybe_merge_siblings(BProps, Obj0),
+    ObjValues = riak_object:get_values(Obj),
+    Reason = get_reason_action(Reason0),
+    get_ops_for_entry_reason(Reason, ObjValues, LI, P, Obj, BKey, ShortPL,
+        Hash, BProps).
+
+-spec get_ops_for_entry_reason(write_reason(), [riak_object:value()],
+        logical_idx(), p(), obj(), bkey(), short_preflist(), hash(),
+        riak_core_bucket:bucket_props()) -> solr_ops().
```
Comment (zeeshanlakhani) [inline src/yz_solrq_helper.erl]: know you're working on a riak_core PR @JeetKunDoug, but just noting the issue w/ this props spec for now.

**Comment by fadushin** [comment]
+1 cbd7f27

**Comment by zeeshanlakhani** [comment]
+1 cbd7f27

**Comment by zeeshanlakhani** [comment]
really good work @JeetKunDoug

**Comment by zeeshanlakhani** [comment]
@borshop merge

**Pull request merged**